python/lsst/obs/base/exposureIdInfo.py

```diff
@@ -21,58 +21,137 @@
 
 __all__ = ["ExposureIdInfo"]
 
+from lsst.daf.butler import DataCoordinate
+from lsst.afw.table import IdFactory
 
-class ExposureIdInfo(object):
-    """Exposure ID and number of bits used.
 
-    Attributes include:
+class ExposureIdInfo:
+    """Struct representing an exposure ID and the number of bits it uses.
 
-    expId
-        exposure ID as an int
-    expBits
-        maximum number of bits allowed for exposure IDs
-    maxBits
-        maximum number of bits available for values that combine exposure ID
-        with other information, such as source ID
-    unusedBits
-        maximum number of bits available for non-exposure info
-        (maxBits - expBits)
+    Parameters
+    ----------
+    expId : `int`
+        Exposure ID.  Note that this is typically the ID of an
+        `afw.image.Exposure`, not the ID of an actual observation, and hence it
+        usually either includes a detector component or is derived from SkyMap
```

Comment:
and will it ever be a `visit` ID combined with detector since below the example uses `visit_detector`?